src/devices/lacrosse_tx22uit.c

```diff
@@ -0,0 +1,133 @@
+/** @file
+    LaCrosse Technology View TX22U-IT temperature, humidity, wind speed/direction
+    and rain sensor.
+
+    Copyright (C) 2020 Caz Yokoyama, caz at caztech dot com
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+/**
+LaCrosse Technology View TX22U-IT temperature, humidity, wind speed/direction
+and rain sensor
+
+Product pages:
+TBW
+
+Specifications:
+TODO: update for TX22U-IT
+- Wind Speed Range: 0 to 178 kmh
+- Degrees of Direction: 360 deg with 16 Cardinal Directions
+- Outdoor Temperature Range: -29 C to 60 C
+- Outdoor Humidity Range: 10 to 99 %RH
+- Update Interval: Every 31 Seconds
+
+http://nikseresht.com/blog/?p=99 tell protocol but my TX22U-IT is different protocol.
+*/
+
+#include "decoder.h"
+
+static int lacrosse_tx22uit_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t const preamble_pattern[] = { 0xaa, 0xaa, 0x2d, 0xd4 };
+
+    data_t *data;
+    uint8_t *b = bitbuffer->bb[0];
+    int offset;
+    int raw_temp = -1, humidity = -1, raw_speed = -1, direction = -1;
+    float temp_c, speed_kmh;
+    int data_valid = 0;
+
+    offset = bitbuffer_search(bitbuffer, 0, 0,
+            preamble_pattern, sizeof(preamble_pattern) * 8);
+
+    if (offset >= bitbuffer->bits_per_row[0]) {
+        if (decoder->verbose) {
+            fprintf(stderr, "%s: Sync word not found\n", __func__);
+        }
+        return DECODE_ABORT_EARLY;
+    }
+
+#if 1 /* TODO: dump bitbuffer only on verbose */
+    bitbuffer_print(bitbuffer);
+#else
+    if (decoder->verbose) {
+        bitbuffer_print(bitbuffer);
+    }
+#endif
+    /*
+       0  1  2  3  4  5  6  7  8  9 10 11 12 13 14
+      aa aa 2d d4 ad 03 05 92 20 01 3d 00 b0 00 00  19.2C  wind 13 at 0.0km/h
+      aa aa 2d d4 ad 02 20 01 40 00 1f 00 00
+      aa aa 2d d4 ad 01 20 01 a3 00 00
+      aa aa 2d d4 ad 02 10 56 20 01 86 00 00        56%
+   */
+    for (offset = 6; offset < bitbuffer->bits_per_row[0] / 8; offset += 2) {
```

Comment:
Shouldn't this offset be the one from bitbuffer_search? Maybe bitbuffer_extract_bytes is needed to align.
If aaaa2dd4 is the full preamble seen, then likely aa2dd4 is a better search, it will cope with missing initial sync preamble bits.
Also starting at byte 6 means skipping that `ad0?`, right? is that the ID and seq?

Reply:
When I move TX22U-IT to garage which is 5m away through floor from side-by-side, I have the following bitbuffer in consistent and stable manner.

[00] {146} aa aa 2d d4 ac 25 05 38 10 60 20 00 3c 00 40 00 a2 00 00 

Its display shows "temperature_C" : 13.800, "humidity" : 61, "wind_avg_km_h" : 0.000, "wind_dir_deg" : 12, rain: 0.

I test with variety of preamble such as a single 0xaa. But "0xaa 0xaa 0x2d 0xd4 0xac, 0x25" gives the most stable data . The last 2 byte may be a device ID. But I can't judge because I only have one TX22U-IT.
Subsequent data seems 2 byte pack. I'll update my code according to this new data.

Reply:
the `aaaa` is a preamble, you should not require the full preamble, just a wide enough trailing portion. That way a few bits missing at the start will still be ok. The `2dd4` is a common sync-word, everything after that is the actual data.
A match on `aa2dd4` will be best. Then use bitbuffer_extract_bytes to get a bitrow with just the aligned data, i.e. b[0] will then be the byte after 2dd4.

Reply:
Done on aa2dd4 bitbuffer_extract_bytes().